### PR TITLE
Initial support for STM32Cube

### PR DESCRIPTION
**Pull request by lealem47**

### CODE REVIEW

File: ide/STM32CUBE/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# wolfSSH for STM32 Cube IDE 
+
+The wolfSSH Cube Pack can be found [here](https://www.wolfssl.com/files/ide/I-CUBE-wolfSSH.pack)
```
Comment (lealem47) [inline ide/STM32CUBE/README.md:3]: Not yet posted on the site. I can provide the pack for review/testing.

File: ide/STM32CUBE/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+1. The first step is to setup the wolfCrypt library in your ST project following the guide here [https://github.com/wolfSSL/wolfssl/blob/master/IDE/STM32Cube/README.md](https://github.com/wolfSSL/wolfssl/blob/master/IDE/STM32Cube/README.md). To run the wolfSSH unit tests, name the entry function `wolfSSHTest` instead of `wolfCryptDemo`.
+
+2. Then install the wolfSSH Cube Pack in the same manner as the wolfSSL pack with CUBEMX.
+
+3. Open the project `.ioc` file and click the `Softare Packs` drop down menu and then `Select Components`. Expand the `wolfSSH` pack and check all the components. 
+
+4. In the `Softare Packs` configuration category of the `.ioc` file, click on the wolfSSH pack and enable the library by checking the box.
```
Comment (dgarske) [inline ide/STM32CUBE/README.md:11]: Spell check file. `Software`

File: wolfssh/test.h [hunk 3]
```diff
@@ -116,14 +116,20 @@
...
     #include <unistd.h>
-    #include <netdb.h>
-    #include <netinet/in.h>
-    #include <netinet/tcp.h>
-    #include <arpa/inet.h>
-    #include <sys/ioctl.h>
     #include <sys/socket.h>
     #include <pthread.h>
-    #include <fcntl.h>
+    #ifdef  WOLFSSH_LWIP
```
Comment (dgarske) [inline wolfssh/test.h:121]: Perhaps this LWIP section should be in its own section above and not in the linux one?

File: wolfssh/port.h [hunk 4]
```diff
@@ -1182,7 +1182,11 @@ extern "C" {
 
 #elif defined(WOLFSSH_USER_FILESYSTEM)
     /* User-defined I/O support */
-    #include "myFilesystem.h"
+    #ifdef WOLFSSH_STM32_CUBEMX
+        #include <wolfssh/myFilesystem.h>
```
Comment (dgarske) [inline wolfssh/port.h:1186]: Could this instead be fixed with an include path adjustment? Did you actually get the filesystem working with the wolfSSH Cube pack?

File: ide/STM32CUBE/README.md (hunk 1 above)
Comment (dgarske) [inline ide/STM32CUBE/README.md:3]: What STM32 hardware platform(s) have you tested with?

**Review by dgarske — CHANGES_REQUESTED** [review]

File: wolfssh/port.h (hunk 4 above)
Comment (lealem47) [reply wolfssh/port.h]: Just building. For the default filesystem stuff, I took the same approach as https://github.com/wolfSSL/wolfssh/tree/master/ide/IAR-EWARM with a dummy filesystem and leave it up to the user to set it up (`WOLFSSH_USER_FILESYSTEM`). The build works with `WOLFSSH_FATFS` as well. As far as running the filesystem tests, they are dependent on `wolfssl/test.h` which isn't in the wolfSSL cube pack. I'm sure there's a way around that but haven't dived deep into it

File: ide/STM32CUBE/README.md (hunk 2 above)
Comment (lealem47) [reply ide/STM32CUBE/README.md]: Fixed.

File: ide/STM32CUBE/README.md (hunk 1 above)
Comment (lealem47) [reply ide/STM32CUBE/README.md:3]: Just the STM32F217 so far

**Review by dgarske — APPROVED** [review]
Looks good so far.

**Pull request merged**